avocado/core/multiplexer.py

```diff
@@ -397,11 +397,8 @@ def parse(self, args):
 
```

Comment:
The wording on the commit message is a bit confusing.  It says: "only the multiplexer plugin should produce variants".  This does *not* match the archicture we settled upon.  It can be seen on the following diagram:

https://github.com/avocado-framework/avocado/pull/1535#issuecomment-253927016

That "Other variant plugin(s)" can provide variants.  Can you better explain the commit message?

Also, WRT to the example given with Avocado-Virt, it would be nice to use the internal component names .  For instance: "this version does the filtering only on the multiplexed tree (`MuxTree`) ... ".  Avocado-Virt can be mentioned as one user of that interface, or even skipped altogether.